pkg/config/bool.go

```diff
@@ -0,0 +1,29 @@
+// Copyright 2021 The PipeCD Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+type TrueByDefaultBool bool
+
+func NewTrueByDefaultBool(v bool) *TrueByDefaultBool {
+	b := TrueByDefaultBool(v)
+	return &b
+}
+
+func (b *TrueByDefaultBool) Value() bool {
+	if b == nil {
+		return true
+	}
+	return bool(*b)
+}
```

Comment:
I understand your thought about adding this new alias to explicitly mention the defaults value for this type, but since we add a new type alias, it may add unnecessary complexity to the codebase, and we have a leak in case there is no value set in the yaml file, if caller side use `bool(*x)` instead of this `x.Value()` the process will be panicked. How about using `*bool` type with a default value is true (using `defaults` package as other places?
This is same test for that method: https://play.golang.org/p/5X2nruNhDZN

Reply:
Nice catch! 
To be honest, I tried that way but it didn't work. And I thought it was a bug from the `defaults` library.
But after checking again, it seems that problem was already resolved and included in the new version.
So let me upgrade that lib to the latest version and try it again.
https://github.com/creasty/defaults/releases

Reply:
Good to hear that 👍